ctapipe/io/dl1eventsource.py

```diff
@@ -0,0 +1,424 @@
+    MorphologyContainer,
+    MCHeaderContainer,
+    MCEventContainer,
+    PeakTimeStatisticsContainer,
+    TimingParametersContainer,
+    TriggerContainer,
+)
+
+
+class DL1EventSource(EventSource):
+    def __init__(
+        self,
+        input_url,
+        config=None,
+        parent=None,
+        **kwargs
+    ):
+        """
+        EventSource for dl1 files in the standard DL1 data format
+
+        Parameters:
+        -----------
+        input_url : str
+            Path of the file to load
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        parent: ??
+            Parent from which the config is used. Mutually exclusive with config
+        kwargs
+        """
+        super().__init__(
+            input_url=input_url,
+            config=config,
+            parent=parent,
+            **kwargs
+        )
+
+        self.file_ = tables.open_file(input_url)
+        self.input_url = input_url
+        self._subarray_info = self._prepare_subarray_info()
+        self._mc_headers = self._parse_mc_headers()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def close(self):
+        self.file_.close()
+
+    @staticmethod
+    def is_compatible(file_path):
+        with open(file_path, 'rb') as f:
+            magic_number = f.read(8)
+        if magic_number != b'\x89HDF\r\n\x1a\n':
+            return False
+        with tables.open_file(file_path) as f:
+            metadata = f.root._v_attrs
+            if 'CTA PRODUCT DESCRIPTION' not in metadata._v_attrnames:
+                return False
+            if metadata['CTA PRODUCT DESCRIPTION'] != 'DL1 Data Product':
+                return False
+        return True
+
+    @property
+    def is_simulation(self):
+        """
+        True for files with a simulation group at the root of the file.
+        """
+        return "simulation" in self.file_.root
+
+    @property
+    def has_simulated_dl1(self):
+        """
+        True for files with telescope-wise event information in the simulation group
+        """
+        if self.is_simulation:
+            if "telescope" in self.file_.root.simulation.event:
+                return True
+        return False
+
+    @property
+    def subarray(self):
+        return self._subarray_info
+
+    @property
+    def datalevels(self):
+        params = 'parameters' in self.file_.root.dl1.event.telescope
+        images = 'images' in self.file_.root.dl1.event.telescope
+        if params and images:
+            return (DataLevel.DL1_IMAGES, DataLevel.DL1_PARAMETERS)
+        elif params:
+            return (DataLevel.DL1_PARAMETERS,)
+        elif images:
+            return (DataLevel.DL1_IMAGES,)
+
+    @property
+    def obs_id(self):
+        return set(self.file_.root.dl1.event.subarray.trigger.col("obs_id"))
```

Comment:
This shouldn't be a set, should it? Or it should be a set every where.

Reply:
Not sure if I understand you correctly. The table includes the obs_id multiple times. For files with only a single obs_id you can just take the first entry, but for merged files this wont work. Or do you just want to cast it back to a list or tuple?

Reply:
I think we have not yet discussed the possibility of having multiple obs ids in the same DL1 file. The current API clearly expects a single int `obs_id`. 

I see we might want to be able to deal with merged files, but I think `obs_id` should be the one of the current event? 

Not sure. Having a set here is unexpected to me.

Reply:
I see the issue.
But for a single event we have `event.index.obs_id` or not?

Reply:
The DL1 format is indeed designed to allow multiple obs_ids in the same file (hence why the simulation config is a table, not header info).  However, the idea was that at least now, you process single runs, and then we would create a "ctapipe-dl1-merge" tool that could join files.  That at least allows the logic to be a bit simpler.    However, I guess that reading the DL1 files will need eventually to support merged files.

To answer the last question: Each event should have a single obs_id, yes. It's the obs_id of the original un-merged file (so the simulation id for MC files) That allows one to uniquely identify an event with (obs_id, event_id).

Reply:
> Therefore having the obs_id property of a DL1EventSource is either:
> 
>     1. not needed, as you always can get the obsid from the event (event.index.obs_id)
>     2. could just be a `set` or `list` 

Actually the base class lists the property `obs_id` as an integer and the name suggests one as well imo.
Currently (with 45a00b4)  there are two properties in the dl1 source: `obs_ids` returning a list and `obs_id` returning the single entry if there is only one id. But that feels a bit hacky as well.

Reply:
Probably we should change it in the base class as well. In principle we can always have multiple merged runs. It should always return a tuple or list, even if a single obs_id, and perhaps the property should be `source.obs_ids` just like `source.datalevels` (which also is a tuple)

Reply:
Agreed.

How does this relate to simtel files? Are there any merged files and would these list both obs ids in the file header?

Reply:
now that we have now, but in principle we could have merged simtel files (we decided not to support them so far in SimTelEventSource, since there were no examples).  I think there exists a tool for simtelarray that can merge them, however.

Reply:
So we just make `obs_id` a tuple/list of one entry for the `SimtelEventSource` and just dont support it for now?